iree/compiler/Conversion/LinalgToLLVM/LinalgBufferizePass.cpp

```diff
@@ -172,6 +173,53 @@ LogicalResult convertAnyLinalgOp(OpBuilder &b, linalg::LinalgOp op,
   return success();
 }
 
+static LogicalResult convertTransferOp(OpBuilder &b,
+                                       VectorTransferOpInterface op,
+                                       BlockAndValueMapping &bvm) {
+  if (op.getShapedType().isa<MemRefType>()) return success();
+  assert(op->getNumResults() == 1);
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = op.getLoc();
+  Value newInputBuffer = bvm.lookup(op.source());
+  if (auto tensorType =
+          op->getResult(0).getType().dyn_cast<RankedTensorType>()) {
+    // If the op return a Tensor allocate a buffer for the returned value.
+    auto tensorShape = tensorType.getShape();
+    auto memrefType = MemRefType::get(tensorShape, tensorType.getElementType());
+    SmallVector<Value, 4> dynOperands;
+    Value outputTensor = op->getResult(0);
+    for (auto dim : llvm::enumerate(tensorShape)) {
+      if (dim.value() == TensorType::kDynamicSize) {
+        dynOperands.push_back(
+            b.create<DimOp>(loc, bvm.lookup(outputTensor), dim.index()));
+      }
+    }
+    auto alloc = b.create<AllocOp>(loc, memrefType, dynOperands);
+    bvm.map(op->getResult(0), alloc);
+    mapAllTieShapeUsesAndReplaceDimUsesOf(op->getResult(0), alloc, bvm);
+  }
+
+  // Replace the tensor operand.
+  if (auto readOp = dyn_cast<vector::TransferReadOp>(op.getOperation())) {
+    auto newRead = b.create<vector::TransferReadOp>(
+        loc, readOp.getVectorType(), newInputBuffer, readOp.indices(),
+        readOp.permutation_map(), readOp.padding(),
+        readOp.masked() ? *readOp.masked() : ArrayAttr());
+    readOp->replaceAllUsesWith(newRead);
+    readOp->erase();
+  } else {
+    auto writeOp = cast<vector::TransferWriteOp>(op.getOperation());
```

Comment:
same here just use `mutableXXX` and do the update inplace.

Reply:
I don't think I can do in-place update since the original transfer_write returns a new tensor that still have uses at this point. Let me know if you think there is a way to do that.

Reply:
You're right, sorry about the noise.